### PR TITLE
hw 2.2 added solution

### PR DESCRIPTION
**Pull request by Vladyslav0711**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,40 @@
...
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int j = matrix.length - 1;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need this variable?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,40 @@
...
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int minElement = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                minElement = matrix[i][j] < minElement ? matrix[i][j] : minElement;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: in this case simple  if condition would look better, Agree?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,40 @@
...
+        }
+        return minElement;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int maxElem = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                maxElem = matrix[i][j] > maxElem ? matrix[i][j] : maxElem;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (Vladyslav0711) [reply src/main/java/core/basesyntax/MatrixOperation.java]: but Math.min() implementation looks like this. Therefore, I thought it was better.
I get it now.

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**